### ISSUE TITLE
ItemAdders Guns

### ISSUE THREAD

**Issue by ZeonBT** [PluginBugs/Issues-ItemsAdder#1107]
Hi there bud, with the new additions from 1.17, like the new particles, items and guis, is it possible for the guns to have an upgrade,. such as a way to colour the particles, set the damage for each different type of gun and to have a sniper have with the new telescope a player could crouch while holding the sniper and when you shot it will fire

**Comment by LoneDev6** [comment]
More requests:
- https://github.com/PluginBugs/Issues-ItemsAdder/issues/2311
- https://github.com/PluginBugs/Issues-ItemsAdder/issues/1443

**Comment by SirCraftypants** [comment]
As i've not kept an eye on your updates for the last few months i'm unaware whether or not you've implemented a new projectile weapon system yet or not.

Just wanted to update you that i was using range and that wasn't working as it should and then i realized i should have been using distance instead so did so and that seems to not work as intended either. Doesn't matter how many blocks away i am, if the guns set to distance 7 and i'm 10 blocks away it still hits the target.

**Comment by LoneDev6** [comment]
> As i've not kept an eye on your updates for the last few months i'm unaware whether or not you've implemented a new projectile weapon system yet or not.

No, it's still marked as future update. This is not a guns plugin so it's not really a priority to deliver an extended guns system other than the already provided basic one.

> Just wanted to update you that i was using range and that wasn't working as it should and then i realized i should have been using distance instead so did so and that seems to not work as intended either. Doesn't matter how many blocks away i am, if the guns set to distance 7 and i'm 10 blocks away it still hits the target.

Please open a separate Github issue page providing all the required information and the item configuration file for me to reproduce the issue.
Do not use the same issue page of another issue/suggestion.

**Comment by 00Hrtz** [comment]
I thought of a short term solution for ammo

What if you could have an event where it checks for the ammo in your inventory and then loads the number required so something like this.

yaml``
    durability:
      max_custom_durability: 2
      custom_durability: 0
      disappear_when_broken: false
    events:
      interact:
        right:
          decrement_durability:
            amount: 2
          play_sound:
            name: minecraft:m41apulse
            volume: 1
            pitch: 1
          play_effect:
            name: SMOKE
          damage_entity_in_sight:
            entity_groups:
              - HOSTILE
              - PLAYERS
            damage: 15
            distance: 12  
        left:
          reload:
            take_ammo: custom_ammo:musket_ball
            amount: 2
            increment_durability: 2
